refract-spec.md

```diff
@@ -137,9 +139,490 @@ This is how it would expressed in the compact version.
+
+In Refract, this number is expanded to a `Number Element`.
+
+```json
+{
+  "element": "number",
+  "attributes": {},
+  "content": 400
+}
+```
+### Boolean Element (Element)
+
+A `Boolean Element` allows for values that are JSON boolean.
+
+#### Properties
+- `element`: boolean (string, fixed)
+- `content` (boolean, required)
+
+#### Examples
+
+In JSON, an example of a boolean is:
+
+```json
+true
+```
+
+In Refract, this boolean is expanded to a `Boolean Element`.
+
+```json
+{
+  "element": "boolean",
+  "attributes": {},
+  "content": true
+}
+```
+
+### Array Element (Element)
+
+An `Array Element` allows for JSON arrays.
+
+#### Properties
+- `element`: array (string, fixed)
+- `content` (array[Element], required)
+
+
+#### Examples
+
+In JSON, an example of an array is:
+
+```json
+["foo", 400, true]
+```
+
+In Refract, this boolean is expanded to a `Boolean Element`.
+
+```json
+{
+  "element": "array",
+  "attributes": {},
+  "content": [
+    {
+      "element": "string",
+      "attributes": {},
+      "content": "foo"
+    },
+    {
+      "element": "number",
+      "attributes": {},
+      "content": 400
+    },
+    {
+      "element": "boolean",
+      "attributes": {},
+      "content": true
+    }
+  ]
+}
+```
+
+### Property Element (Element)
+
+A `Property Element` allows for values that MUST be used in `Object Element` elements.
+
+#### Properties
+- `element`: property (string, fixed)
+- `attributes`
+    - `name` (string, required) - Name of property
+- `content` (Element, required) - Any element that extends Element
+
+#### Examples
+
+To see how this element is used, view the `Object Element` definition.
+
+### Object Element (Element)
+
+A `Object Element` allows for values that are JSON boolean.
+
+#### Properties
+- `element`: object (string, fixed)
+- `content` (array[Property Element], required)
+
+#### Examples
+
+In JSON, an example of an object is:
+
+```json
+{
+  "foo": "bar"
+}
+```
+
+In Refract, this object is expanded to a `Object Element`.
+
+```json
+{
+  "element": "object",
+  "attributes": {},
+  "content": [
+    {
+      "element": "property",
+      "attributes": {
+        "name": "foo"
+      },
+      "content": {
+        "element": "string",
+        "attributes": {},
+        "content": "bar"
+      }
+    }
+  ]
+}
+```
+
+## Referencing Elements
+
+The referencing elements are provided as part of the base specification for the purpose of identifying and referencing elements.
+
+### ID (Element)
+
+The `id` element is a way to identify an element in a document. IDs MUST be unique throughout a document.
+
+#### Example
+
+```json
+{
+  "element": "string",
+  "attributes": {
+    "id": "foo"
+  },
+  "content": "bar"
+}
+```
+
+### Ref (Element)
+
+The `ref` element MAY be used to reference elements in remote documents or elements in the local document.
+
+1. If the content is an absolute URL with a fragment identifier, the fragment identifer references the ID in the corresponding document.
```

Comment:
Nit: Since you are discussing `content` property I would say this list should be moved under the `content`